orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -122,7 +136,7 @@ IMPLEMENT_GRADIENT_BUILDER(GetMatMulGradient) {
     AttributeProto transpose_second_input = MakeAttribute("transB", int64_t(1));
 
     if (A_shape.size() == 2 && B_shape.size() == 2) {
-      NodeDef zero_constant_node = ZeroConstantNode();
+      NodeDef zero_constant_node = ZeroConstantNode(OElemType(0));
```

Comment:
Matmul's inputs and output should share the same type. 

So all the type selection in this function can share the same type parameter...